lib/Default/SmrPlanet.class.inc

```diff
@@ -641,26 +640,31 @@ class SmrPlanet {
 	}
 
 	public function getMaxBuildings($buildingTypeID=false) {
-		if (!isset($this->maxBuildings)) {
-			$this->setMaxBuildings();
-		}
+		// Make sure the data is populated
+		$this->setMaxBuildings();
 		if($buildingTypeID===false) {
 			return $this->maxBuildings;
 		}
-		
-		if (isset($this->maxBuildings[$buildingTypeID])) {
-			return $this->maxBuildings[$buildingTypeID];
-		}
-		return 0;
+		return $this->maxBuildings[$buildingTypeID];
 	}
 	
-	public function setMaxBuildings() {
-		$this->maxBuildings = array();
-
-		$this->db->query('SELECT construction_id, max_amount FROM planet_can_build WHERE planet_type_id='.$this->getTypeID());
-		while ($this->db->nextRecord()) {
-			$this->maxBuildings[$this->db->getInt('construction_id')] = $this->db->getInt('max_amount');
+	/**
+	 * Populates the `maxBuildings` attribute from the `planet_can_build` database table.
+	 */
+	private function setMaxBuildings() {
```

Comment:
Could you rename this to `update` or `fetch` or something - `set` is pretty confusing alongside functions like `setShields` (the same goes for the other `set` functions that just fetch data)

Reply:
You're 100% right. I was trying to fit into the naming scheme of the class, but it was already inconsistent, so I guess I should take a moment to rename a few things.

I'm thinking:
* calculateInterest -> matureBond (sounds a bit weird, maybe payoutBond?)
* setMaxBuildings -> initializeMaxBuildings
* updateTypeInfo -> initializeTypeInfo
* getOptions -> getMenuOptions
* setOptions -> initializeMenuOptions

Look okay to you?

Reply:
I think shortening it to `initX` or `fetchX` would be good, you could also have it directly inside the `getX` methods as well which is common in a bunch of other places.  For `calculateInterest` I think it's an ok name, but something like `checkBondMaturity` could be better